Source/Request.swift

```diff
@@ -327,15 +328,23 @@ extension Request: CustomDebugStringConvertible {
         }
 
         if let headerFields = request.allHTTPHeaderFields {
-            for (field, value) in headerFields where field != "Cookie" {
-                headers[field] = value
+            for (field, value) in headerFields {
+                if field != "Cookie" {
+                    headers[field] = value
+                } else {
+                    cookies = value
+                }
             }
         }
 
         for (field, value) in headers {
             components.append("-H \"\(field): \(value)\"")
         }
 
+        if let c = cookies, !session.configuration.httpShouldSetCookies {
```

Comment:
I think this can simply be added as an else to the cookie logic that ends on line 319.

```swift
if session.configuration.httpShouldSetCookies {
    if
        let cookieStorage = session.configuration.httpCookieStorage,
        let cookies = cookieStorage.cookies(for: url), !cookies.isEmpty
    {
        let string = cookies.reduce("") { $0 + "\($1.name)=\($1.value);" }

    #if swift(>=3.2)
        components.append("-b \"\(string[..<string.index(before: string.endIndex)])\"")
    #else
        components.append("-b \"\(string.substring(to: string.characters.index(before: string.endIndex)))\"")
    #endif
    }
} else if
    let sessionCookies = session.configuration.httpAdditionalHeaders?["Cookie"] as? String,
    let requestCookies = request.allHTTPHeaderFields?["Cookie"] {
    components.append("-b \"\(sessionCookies);\(requestCookies)\"")
}
```

With that, none of the other changes are needed. 

Also, is it possible for a user to use both the standard cookie storage and attach their own cookies, or would one overwrite the other? Perhaps this code should access the storage cookies, if available, as well as the session and request cookies, regardless of the `session.configuration.httpShouldSetCookies` setting?